utils/Felt_test.go

```diff
@@ -0,0 +1,39 @@
+package utils
+
+import (
+	"testing"
+)
+
+func TestByteArrToFelt(t *testing.T) {
+	var tests = []struct {
+		in  string
+		out []string
+	}{
+		{
+			in:  "hello",
+			out: []string{"0x0", "0x68656c6c6f", "0x5"},
```

Comment:
I would have expected the first string to be 0x1 (similarly for the next test case, I would have expected 0x2)

Reply:
Based on the example as mentioned in the [doc](https://docs.starknet.io/architecture-and-concepts/smart-contracts/serialization-of-cairo-types/#serialization_of_byte_arrays)

First element in the array indicates number of words with 31 bytes of length
```1, // Number of 31-byte words in the array construct.
    0x4c6f6e6720737472696e672c206d6f7265207468616e203331206368617261, // 31-byte word.
    0x63746572732e, // Pending word
    6 // Length of the pending word, in bytes```